builtin/providers/template/resource.go

```diff
@@ -0,0 +1,119 @@
+package template
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hashicorp/terraform/config"
+	"github.com/hashicorp/terraform/config/lang"
+	"github.com/hashicorp/terraform/config/lang/ast"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resource() *schema.Resource {
+	return &schema.Resource{
+		Create: Create,
+		Read:   Read,
+		Update: Update,
+		Delete: Delete,
+		Exists: Exists,
+
+		Schema: map[string]*schema.Schema{
+			"filename": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "file to read template from",
+			},
+			"vars": &schema.Schema{
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Default:     make(map[string]interface{}),
+				Description: "variables to substitute",
+			},
+			"rendered": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Default:     nil,
+				Description: "rendered template",
+			},
+		},
+	}
+}
+
+func Create(d *schema.ResourceData, meta interface{}) error { return eval(d) }
+func Update(d *schema.ResourceData, meta interface{}) error { return eval(d) }
+func Read(d *schema.ResourceData, meta interface{}) error   { return nil }
+func Delete(d *schema.ResourceData, meta interface{}) error {
+	d.SetId("")
+	return nil
+}
+func Exists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	// Reload every time in case something has changed.
+	// This should be cheap, and cache invalidation is hard.
+	return false, nil
+}
```

Comment:
Hey @josharian - working on #1866 and coming back around to this. I think we need to rework the lifecycle a bit here. Will follow up with a PR.